packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardTitle.base.tsx

```diff
@@ -9,66 +9,65 @@ const getClassNames = classNamesFunction<IDocumentCardTitleStyleProps, IDocument
 export interface IDocumentCardTitleState {
   truncatedTitleFirstPiece?: string;
   truncatedTitleSecondPiece?: string;
+  clientWidth?: number;
+  previousTitle: string;
+  needMeasurement: boolean;
 }
 
-const TRUNCATION_SEPARATOR = '&hellip;';
-const TRUNCATION_MINIMUM_LENGTH = 40; // This is the length we know can fit into the min width of DocumentCard.
-const TRUNCATION_MAXIMUM_LENGTH = 90 - TRUNCATION_SEPARATOR.length;
-
-// This is the length we know can fit into the min width 2 lines of DocumentCard.
-const TRUNCATION_MINI_LENGTH_SECONDARY = 80;
-const TRUNCATION_MAX_LENGTH_SECONDARY = 130 - TRUNCATION_SEPARATOR.length;
-const TRUNCATION_FIRST_PIECE_LONGER_BY = 10;
 const TRUNCATION_VERTICAL_OVERFLOW_THRESHOLD = 5;
 
 export class DocumentCardTitleBase extends BaseComponent<IDocumentCardTitleProps, IDocumentCardTitleState> {
   private _titleElement = React.createRef<HTMLDivElement>();
+  private _measureTitleElement = React.createRef<HTMLDivElement>();
+
   private _scrollTimerId: number;
-  private _truncatedTitleAtWidth: number;
-  private _isTruncated: boolean;
   private _classNames: IProcessedStyleSet<IDocumentCardTitleStyles>;
 
   constructor(props: IDocumentCardTitleProps) {
     super(props);
 
     this.state = {
       truncatedTitleFirstPiece: '',
-      truncatedTitleSecondPiece: ''
+      truncatedTitleSecondPiece: '',
+      previousTitle: props.title,
+      needMeasurement: !!props.shouldTruncate
     };
   }
 
-  public componentDidMount(): void {
-    const { title, shouldTruncate, showAsSecondaryTitle } = this.props;
-    const miniLength = showAsSecondaryTitle ? TRUNCATION_MINI_LENGTH_SECONDARY : TRUNCATION_MINIMUM_LENGTH;
-    if (shouldTruncate && title && title.length > miniLength) {
-      if (this._doesTitleOverflow()) {
-        this._startTruncation(this.props);
-      }
-      this._events.on(window, 'resize', this._updateTruncation);
+  public componentDidUpdate(): void {
+    if (this.props.title !== this.state.previousTitle) {
```

Comment:
In general, a better solution would just be to have a `TruncatedText` component which can take in a `rows` prop to abstract the text truncation. Then the DocumentCard itself won't get re-rendered, but only the text within it.

Reply:
It could potentially even use ResizeObserver rather than window resize, which would catch more scenarios.

Reply:
ResizeObserver is not supported in edge and IE?